### DISCUSSION TITLE
Is Tide still under active development?

### DISCUSSION THREAD

**Discussion by maxcountryman** [http-rs/tide#888] [Q&A] [answered]
Hi all, I'm curious if one of the maintainers might be able to comment on the status of this project and what the future ahead looks like?

**Comment by yoshuawuyts** [comment, marked as answer]
Heya, yeah that's a good question! - I guess the practical short answer is: no, Tide is not very actively being developed right now. The design of Tide is my responsibility, and my attention for the past year has mostly gone into developing the underlying Rust features required to make projects like Tide work better in the future.

Now for the longer answer: just because we're not actively pushing out features doesn't mean the project is abandoned, and we have no plans to abandon it either. We're still merging maintenance patches, and folks (including myself) are successfully using it in projects. From my perspective the Rust async ecosystem is still in a fledgling state, and in order for projects like Tide to soar we need to improve the underlying fundamentals first. So that's where why my focus is there right now.

For example, one of those fundamentals will be figuring out async traits, which will likely lead us to rewrite `AsyncRead`/`AsyncWrite`/`AsyncIterator` to be `async fn`-based. In order to prove that this is a feasible approach, the Rust Async WG is planning to do PoC rewrite of `async-h1` and other layers entirely without `poll`-based state machines. To prove that we can reimplement an entire HTTP stack without manual `poll`ing. If this works out, that'll be a huge deal for async Rust in general - and in turn would also conveniently leave `http-rs` with a greatly simplified HTTP stack.

Tide and `http-rs` have always served a dual purpose: `tide` was initially created to (among other things) evaluate the ergonomics of `async/.await` as part of Rust Network Services WG. But we quickly realized that we not only wanted to have it be a PoC, we also wanted folks to use it in their projects. In a way, the temporary re-focus on the underlying Rust primitives is par the course for the project. The focus might not be directly on Tide right now, but that doesn't mean we're not making progress on it either.

I hope that somewhat answers your question!

**Reply by sirth1337** [reply]
Hi @yoshuawuyts I've really enjoyed using tide and taken a great in depth analysis at it https://nyxtom.dev/2022/06/26/tide/, submitted a few different middleware (tide-flash, tide-jwt) and recently went through and did some refactoring to improve ergonomics (https://github.com/http-rs/tide/pull/895) I'd love to help maintain this project if possible!

**Reply by yoshuawuyts** [reply]
@nyxtom hi, that's amazing! -- I'm a bit short on time right now, so I'm not always around to reply to issues. But I'm excited you're excited for Tide! If you hop into the [http-rs zulip](https://http-rs.zulipchat.com/join/urc4hdub4jkzip6ubt43mcyi/), maybe we can chat more there? Thanks!

**Reply by Jzow** [reply]
Hi @yoshuawuyts I have to admit that [summer boot](https://github.com/summer-os/summer-boot) is the development of fork tide. I want to integrate some components of automatic scanning and automatic configuration. I think tide does not support http2 at present, and I hope it can be improved in the future

**Reply by kmf-lab** [reply]
Can you give us an update now that we are in 2024 and we have 1.75?

**Reply by Fishrock123** [reply]
I would consider tide now somewhere between "not under active development" and "abandoned". Maybe it will pick back up in the future. Personally I am using other alternatives at my current workplace.

**Reply by mdtusz** [reply]
Is there any chance a "maintenance" release could get made soonish? There's a fair number of maintenance-type PR's that have been in wait for a long time that aren't necessarily blocking usage, but are thorn-in-the-side type things (e.g. [adding the route to the request](https://github.com/http-rs/tide/pull/899), [resetting middleware](https://github.com/http-rs/tide/pull/855) and many others).

Unrelated to the main tide crate, but it would be awesome if the related async-session crates could get some dependency bumps as well ([particularly around redis](https://github.com/jbr/async-redis-session/pull/27)...).

FWIW, we're using tide in a "production" use-case so I would be very happy to contribute some more time towards maintenance type work in tide and the related ecosystem so that we don't need to jump ship to Axum and deal with all the headaches that will bring.

**Reply by maxcountryman** [reply]
Just to chime in here re async-session: I had offered several times to help with that crate but ultimately my sense is there isn't appetite for it and I ended up implementing my own session to unblock my work.
